crates/relayer-types/src/core/ics04_channel/packet.rs

```diff
@@ -178,6 +181,32 @@ impl Packet {
 
         height_timed_out || timestamp_timed_out
     }
+
+    pub fn validate_fields(
+        &self,
+        max_memo_size: usize,
+        max_receiver_size: usize,
+    ) -> Result<(usize, usize), Error> {
+        let any_packet: RawPacketData =
+            serde_json::from_slice(&self.data).map_err(Error::serde_json_error)?;
+        let packet: Ics20PacketData = any_packet
+            .try_into()
+            .map_err(|e: Ics20Error| Error::decode_ics20_packet(e.to_string()))?;
```

Comment:
This is probably overkill, what do you think about just looking at the fields of the RawPacketData instead?

Reply:
Good point I will remove this